docker/resource_docker_config_test.go

```diff
@@ -0,0 +1,100 @@
+package docker
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDockerConfig_basicNotUpdateable(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		// swarm will be initialized in 'testAccPreCheck' if necessary
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckDockerConfigDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: `
+				resource "docker_config" "foo" {
+					name = "foo-config"
+					data = "Ymxhc2RzYmxhYmxhMTI0ZHNkd2VzZA=="
+				}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("docker_config.foo", "name", "foo-config"),
+					resource.TestCheckResourceAttr("docker_config.foo", "updateable", "false"),
+					resource.TestCheckResourceAttr("docker_config.foo", "data", "Ymxhc2RzYmxhYmxhMTI0ZHNkd2VzZA=="),
+				),
+			},
+		},
+	})
+}
+func TestAccDockerConfig_basicUpdateable(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		// swarm will be initialized in 'testAccPreCheck' if necessary
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckDockerConfigShouldStillExist,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: `
+				resource "docker_config" "foo" {
```

Comment:
The test configuration should be it's own method that receives a randomized string or int. See `TestAccAWSLaunchConfiguration_updateRootBlockDevice ` for an example:

- https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_launch_configuration_test.go#L130

Reply:
Got it. Like refactoring the tests to keep them DRY